data/clean.ts

```diff
@@ -84,19 +76,107 @@ async function parseAndIndexGrades(): Promise<
         subjectIndex[subjectArea][catalogNumber] = [];
       }
       subjectIndex[subjectArea][catalogNumber].push(row);
+
+      const summaryKey = `${enrollmentTerm}|${subjectArea}|${catalogNumber}|${sectionNumber}`;
+      if (!summaryIndex[summaryKey]) {
+        summaryIndex[summaryKey] = [];
+      }
+      summaryIndex[summaryKey].push(row);
     });
 
   return new Promise((resolve) => {
     parser.on("end", () => {
-      resolve([rows, instructorIndex, subjectIndex]);
+      resolve([rows, instructorIndex, subjectIndex, summaryIndex]);
     });
   });
 }
 
+function calculateSummaries(summaryIndex: SummaryIndex): SummaryRow[] {
+  const GRADE_VALUES = {
+    "A+": 4,
+    A: 4,
+    "A-": 3.7,
+    "B+": 3.3,
+    B: 3,
+    "B-": 2.7,
+    "C+": 2.3,
+    C: 2,
+    "C-": 1.7,
+    "D+": 1.3,
+    D: 1,
+    "D-": 0.7,
+    F: 0,
+  };
```

Comment:
Could be worthwhile to move this into `utils.ts` and create a function `getGradeValue`. Then we could use that function to improve the `compareGrades` function (also in `utils.ts`) which right now just compares the grades based on their position in a fixed array